### PR TITLE
Insitu temp and salt

### PR DESCRIPTION
**Pull request by guillaumevernieres**
Mostly Shastri's work.

This will need a companion pr in the global-workflow.

### CODE REVIEW

**Comment by guillaumevernieres** [comment]
I only looked at the hera log:
```
[[38;5;39m2024-05-30 22:58:33,149 - DEBUG    - atm_analysis: ( <pygfs.task.atm_analysis.AtmAnalysis object at 0x148ad883fdc0> )^[[0m                                                                                      
^[[38;21m2024-05-30 22:58:33,149 - INFO     - analysis    : BEGIN: pygfs.task.analysis.get_jedi_config^[[0m                                                                                                                
^[[38;5;39m2024-05-30 22:58:33,149 - DEBUG    - analysis    : ( <pygfs.task.atm_analysis.AtmAnalysis object at 0x148ad883fdc0> )^[[0m                                                                                      
^[[38;21m2024-05-30 22:58:33,149 - INFO     - analysis    : Generate JEDI YAML config: /scratch1/NCEPDEV/da/Cory.R.Martin/CI/GDASApp/workflow/PR/1135/global-workflow/sorc/gdas.cd/build/gdas/test/atm/global-workflow/tes\
trun/RUNDIRS/gdas_test/gdasatmanl_18/gdas.t18z.atmvar.yaml^[[0m                                                                                                                                                            
Traceback (most recent call last):                                                                                                                                                                                         
  File "/scratch1/NCEPDEV/da/Cory.R.Martin/CI/GDASApp/workflow/PR/1135/global-workflow/scripts/exglobal_atm_analysis_initialize.py", line 24, in <module>                                                                  
    AtmAnl.initialize()                                                                                                                                                                                                    
  File "/scratch1/NCEPDEV/da/python/gdasapp/wxflow/20240307/src/wxflow/logger.py", line 266, in wrapper                                                                                                                    
    retval = func(*args, **kwargs)                                                                                                                                                                                         
  File "/scratch1/NCEPDEV/da/Cory.R.Martin/CI/GDASApp/workflow/PR/1135/global-workflow/ush/python/pygfs/task/atm_analysis.py", line 74, in initialize                                                                      
    super().initialize()                                                                                                                                                                                                   
  File "/scratch1/NCEPDEV/da/Cory.R.Martin/CI/GDASApp/workflow/PR/1135/global-workflow/ush/python/pygfs/task/analysis.py", line 36, in initialize                                                                          
    self.task_config.jedi_config = self.get_jedi_config()                                                                                                                                                                  
  File "/scratch1/NCEPDEV/da/python/gdasapp/wxflow/20240307/src/wxflow/logger.py", line 266, in wrapper                                                                                                                    
    retval = func(*args, **kwargs)                                                                                                                                                                                         
  File "/scratch1/NCEPDEV/da/Cory.R.Martin/CI/GDASApp/workflow/PR/1135/global-workflow/ush/python/pygfs/task/analysis.py", line 82, in get_jedi_config                                                                     
    jedi_config = render(jcb_config)                                                                                                                                                                                       
  File "/scratch1/NCEPDEV/da/Cory.R.Martin/CI/GDASApp/workflow/PR/1135/global-workflow/ush/python/jcb/renderer.py", line 265, in render                                                                                    
    return jcb_object.render(algorithm)                                                                                                                                                                                    
  File "/scratch1/NCEPDEV/da/Cory.R.Martin/CI/GDASApp/workflow/PR/1135/global-workflow/ush/python/jcb/renderer.py", line 221, in render                                                                                    
    for observer in observers['observers']:                                                                                                                                                                                
TypeError: list indices must be integers or slices, not str 
```
is this expected @RussTreadon-NOAA ?

**Comment by guillaumevernieres** [comment]
@RussTreadon-NOAA , @CoryMartin-NOAA , @danholdaway 
Once this is merged, I'm going to build from scratch and run the ctest after updating the `jcb` and `gdas.cd` # in the companion `g-w` PR https://github.com/NOAA-EMC/global-workflow/pull/2644
@danholdaway , is the HEAD of `jcb` develop OK as the new #?

File: parm/soca/obs/obs_list.yaml [hunk 1]
```diff
@@ -3,30 +3,30 @@ observers:
...
+- !INC ${OBS_YAML_DIR}/sst_viirs_n20_l3u.yaml
+- !INC ${OBS_YAML_DIR}/sst_abi_g16_l3c.yaml
+- !INC ${OBS_YAML_DIR}/sst_abi_g17_l3c.yaml
+- !INC ${OBS_YAML_DIR}/sst_ahi_h08_l3c.yaml
 
 # Ice concentration
 - !INC ${OBS_YAML_DIR}/icec_amsr2_north.yaml
 - !INC ${OBS_YAML_DIR}/icec_amsr2_south.yaml
 
 # in situ: monthly
```
Comment (apchoiCMD) [inline parm/soca/obs/obs_list.yaml:19]: love this part now in-

**Review by apchoiCMD — APPROVED** [review]
looks good to me-

**Review by AndrewEichmann-NOAA — COMMENTED** [review]
Looks ok

**Pull request merged**